springboot/src/main/java/ca/jrvs/apps/trading/dao/JdbcCrudDao.java

```diff
@@ -0,0 +1,159 @@
+package ca.jrvs.apps.trading.dao;
+
+import ca.jrvs.apps.trading.model.domain.Entity;
+import ca.jrvs.apps.trading.model.domain.Quote;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DataRetrievalFailureException;
+import org.springframework.dao.IncorrectResultSizeDataAccessException;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.BeanPropertySqlParameterSource;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+
+public abstract class JdbcCrudDao<T extends Entity<Integer>> implements CrudRepository<T, Integer> {
+
+    private static final Logger logger = LoggerFactory.getLogger(JdbcCrudDao.class);
+
+    abstract public JdbcTemplate getJdbcTemplate();
+
+    abstract public SimpleJdbcInsert getSimpleJdbcInsert();
+
+    abstract public String getTableName();
+
+    abstract public String getIdColumnName();
+
+    abstract Class<T> getEntityClass();
+
+    @Override
+    public <S extends T> S save(S entity){
+        if (existsById(entity.getId())){
+            if (updateOne(entity)!= 1){
+                throw new DataRetrievalFailureException("Unable to update quote.");
+            }
+        } else {
+            addOne(entity);
+        }
+        return entity;
+    }
+    @Override
+    public <S extends T> Iterable<S> saveAll(Iterable<S> entities) {
+        List<S> out = new ArrayList<>();
+        for (S q : entities) {
+            out.add((S) save(q));
+        }
+        return out;
+    }
+    private <S extends T> void addOne(S entity){
+        SqlParameterSource parameterSource = new BeanPropertySqlParameterSource(entity);
+        Object newId = getSimpleJdbcInsert().executeAndReturnKey(parameterSource);
+        entity.setId((Integer) newId);
+    }
+
+
+    abstract public int updateOne(T entity);{
```

Comment:
- fix your coding style. you don't need {}. `abstract public int updateOne(T entity);`
- remove extra newlines (e.g. line 58). it looks weird when you have inconsistent newlines
- autoformat source code files in batch https://www.notion.so/jarviscanada/IntelliJ-IDEA-Guide-2e124ee013ee4ef3b8e1ee55e3b494aa#ec04f4fa69974f5facf996fae8a85d4d